File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        n = len(nums)
        count = 0

        prefixsum = [0] * n
        prefixsum[0] = nums[0]

        for i in range(1, n):
            prefixsum[i] = prefixsum[i - 1] + nums[i]

        for i in range(n - 1):
            if prefixsum[i] >= prefixsum[-1] - prefixsum[i]:
                count += 1

        return count

File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var waysToSplitArray = function (nums) {
    const n = nums.length;
    const prefixsum = new Array(n).fill(0);
    prefixsum[0] = nums[0];

    for (let i = 1; i < n; i++) {
        prefixsum[i] = prefixsum[i - 1] + nums[i]
    }

    let count = 0;

    for (let i = 0; i < n - 1; i++) {
        if (prefixsum[i] >= prefixsum[n - 1] - prefixsum[i]) {
            count++;
        }
    }

    return count;
};